azuki-core/src/main/kotlin/com/anaplan/engineering/azuki/core/runner/MultiOracleScenarioRunner.kt

```diff
@@ -208,24 +201,46 @@ class MultiOracleScenarioRunner<
 
     class OracleResult<
         AF : ActionFactory,
+        CF : CheckFactory,
         QF : QueryFactory,
         AGF : ActionGeneratorFactory
         >(
         val verified: Boolean,
-        val instanceName: String,
-        val instanceVersion: String?,
+        val start: LocalDateTime,
+        val instance: ImplementationInstance<AF, CF, QF, AGF>,
         val taskResults: List<TaskResult<OracleScenario<AF, QF, AGF>, *>>
-    )
+    ) {
+        class Builder<
+            AF : ActionFactory,
+            CF : CheckFactory,
+            QF : QueryFactory,
+            AGF : ActionGeneratorFactory
+            >(
+            val instance: ImplementationInstance<AF, CF, QF, AGF>,
+            val start: LocalDateTime = LocalDateTime.now()
+        ) {
+            private val taskResults = mutableListOf <TaskResult<OracleScenario<AF, QF, AGF>, *>>()
+
+            fun add(taskResult: TaskResult<OracleScenario<AF, QF, AGF>, *>) = taskResults.add(taskResult)
+
+            fun addAll(taskResults: List<TaskResult<OracleScenario<AF, QF, AGF>, *>>) = this.taskResults.addAll(taskResults)
+
+            fun build(verified: Boolean = false) = OracleResult(verified, start, instance, taskResults)
```

Comment:
I understand that in `verifyWithOracle` we always know if the scenario is verified or not when we call this, but to me it would make more sense if the the value of `verified` was set based on the stored task results, e.g.:
```suggestion
            fun build() = OracleResult(taskResults.all { it.verified }, start, instance, taskResults)
```

Reply:
but the stored task results
a) don't have a verified property
b) do not have the concept of being verified or not

Reply:
Ah, I misread it. I thought they were instances of https://github.com/anaplan-engineering/azuki/blob/bd7b4574888bc4ec3efff6f292be89e7a4ac90cd/azuki-core/src/main/kotlin/com/anaplan/engineering/azuki/core/runner/MultiOracleScenarioRunner.kt#L232
rather than https://github.com/anaplan-engineering/azuki/blob/bd7b4574888bc4ec3efff6f292be89e7a4ac90cd/azuki-core/src/main/kotlin/com/anaplan/engineering/azuki/core/runner/TaskResult.kt#L5